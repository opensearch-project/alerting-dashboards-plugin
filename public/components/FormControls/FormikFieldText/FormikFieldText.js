/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiCompressedFieldText } from '@elastic/eui';

import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikFieldText = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <FieldText name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <FieldText name={name} form={form} field={field} inputProps={inputProps} />
      )
    }
  />
);

const FieldText = ({
  name,
  form,
  field,
  inputProps: { onChange, isInvalid, onFocus, ...rest },
}) => (
  <EuiCompressedFieldText
    {...field}
    {...rest}
    onChange={(e) =>
      typeof onChange === 'function' ? onChange(e, field, form) : field.onChange(e)
    }
    onFocus={typeof onFocus === 'function' ? (e) => onFocus(e, field, form) : onFocus}
    isInvalid={typeof isInvalid === 'function' ? isInvalid(name, form) : isInvalid}
  />
);

FormikFieldText.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

FieldText.propTypes = {
  name: PropTypes.string.isRequired,
  inputProps: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikFieldText;
