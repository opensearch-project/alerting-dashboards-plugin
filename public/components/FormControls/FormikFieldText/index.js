/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikFieldText from './FormikFieldText';

export default FormikFieldText;
