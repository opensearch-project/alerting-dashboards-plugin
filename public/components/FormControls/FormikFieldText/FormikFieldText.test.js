/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import FormikFieldText from './FormikFieldText';

describe('FormikFieldText', () => {
  test('renders', () => {
    const component = (
      <Formik>
        <FormikFieldText name="testing" />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
