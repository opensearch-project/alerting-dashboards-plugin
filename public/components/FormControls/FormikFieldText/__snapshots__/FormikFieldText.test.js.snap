// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikFieldText renders 1`] = `
<div
  class="euiFormControlLayout euiFormControlLayout--compressed"
>
  <div
    class="euiFormControlLayout__childrenWrapper"
  >
    <input
      class="euiFieldText euiFieldText--compressed"
      name="testing"
      type="text"
    />
  </div>
</div>
`;
